### PR TITLE
Add Notification module

### PR DESCRIPTION
**Pull request by diogo-mendes**
In this PR, we add the Notification module to the diana design, which includes the `Notification` and `NotificationStack` components.

For detailed info please check the storybook.

### CODE REVIEW

File: src/modules/Notification/Notification.story.mdx [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+      <td>Style of the Notification title</td>
+    </tr>
+    <tr>
+      <td>wrapper</td>
+      <td></td>
+      <td>Style of the wrapper element</td>
+    </tr>
+    <tr>
+      <td></td>
+      <td>.enter</td>
```
Comment (diogo-mendes) [inline src/modules/Notification/Notification.story.mdx]: A more detailed description of these 4 selectors would be cool if you guys have any suggestions.

Comment (acaldas) [reply src/modules/Notification/Notification.story.mdx]: I think it's understandable as it is, it follows the pattern of animations on React

File: src/modules/Notification/Notification.story.mdx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        Style of the wrapper element immediately (1 tick) after ".exit" is
+        added.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+# NotificationStack
+
+The `NotificationStack` receives an array of notifications and handles their display. Through the theme, the aninmations applied to showing and hiding a notification can be configured.
```
Comment (acaldas) [inline src/modules/Notification/Notification.story.mdx]: animations typo :neckbeard:

**Comment by acaldas** [comment]
We should also have a way to handle state for global notifications, so that a notification can be easily triggered by any component and be consumed by the NoticationStack. We could use React Context for this.

**Comment by diogo-mendes** [comment]
I agree with you Afonso, but that can also be achieved by the `NotificationStack`s parent component right? He receives the context and passes the appropriate notifications to the `NotificationStack`. That way we keep this component purely presentational 🙂

**Pull request merged**